demo/cpp/main.cpp

```diff
@@ -0,0 +1,288 @@
+#include <chrono>
+#include <iostream>
+#include <fstream>
+#include <iomanip>
+#include <NvUffParser.h>
+#include <cuda_runtime_api.h>
+
+#include <opencv2/core/core.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+#include <opencv2/imgcodecs/imgcodecs.hpp>
+
+#include <plugin/amirInferPlugin.h>
+#include "data.h"
+#include "inout_transform.h"
+
+
+//  Some required utilities
+class Logger : public nvinfer1::ILogger {
+public:
+    void log(Severity severity, const char* msg) override {
+        // remove this 'if' if you need more logged info
+        if ((severity == Severity::kERROR) || (severity == Severity::kINTERNAL_ERROR)) {
+            std::cout << msg << std::endl;
+        }
+    }
+} gLogger;
+
+// Create definition TRTUniquePtr for unique pointer of TensorRT’s classes
+// destroy TensorRT objects if something goes wrong
+struct TRTDestroy {
+    template< class T >
+    void operator()(T* obj) const {
+        if (obj) {
+            obj->destroy();
+        }
+    }
+};
+
+template< class T >
+using TRTUniquePtr = std::unique_ptr< T, TRTDestroy >;
+
+size_t getSizeByDim(const nvinfer1::Dims& dims) {
+    size_t size = 1;
+    for (size_t i = 0; i < dims.nbDims; ++i) size *= dims.d[i];
+    return size;
+}
+
+//-----------------------------------------------------------------------------------------------------
+
+// Load the model function
+void parseUffModel(const std::string& model_path, int batch_size, TRTUniquePtr<nvinfer1::ICudaEngine>& engine,
```

Comment:
Has this function be used here? If not, please remove the unnecessary functions.